password_enhancements.module

```diff
@@ -0,0 +1,471 @@
+  _password_enhancements_add_password_requirement_validation($form, $form_state);
+}
+
+/**
+ * Adds password requirement validation.
+ *
+ * @param array $form
+ *   The form for which the password requirements validation should be added.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current form state.
+ */
+function _password_enhancements_add_password_requirement_validation(array &$form, FormStateInterface $form_state): void {
+  if (empty($form['account']['pass'])) {
+    /** @var \Drupal\Core\Config\Config $config */
+    $config = Drupal::config('password_enhancements.settings');
+    if (empty($config->get('require_password'))) {
+      // No password field available, nothing to do here.
+      return;
+    }
+
+    $form['account']['pass'] = [
+      '#type' => 'password_confirm',
+      '#size' => 25,
+      '#description' => t('Provide a password for the new account in both fields.'),
+      '#required' => TRUE,
+    ];
+    $form['actions']['submit']['#submit'][] = 'password_enhancements_form_user_register_form_alter_password_required_submit';
+  }
+
+  // Add weight for account fields.
+  $weight = 0;
+  foreach (Element::children($form['account']) as $key) {
+    if (!array_key_exists('#weight', $form['account'][$key])) {
+      $form['account'][$key]['#weight'] = $weight++;
+    }
+  }
+
+  // Get roles.
+  $roles = $form_state->getValue('roles') ?? [];
+  $input = $form_state->getUserInput();
+  // If there was no input then grab the default roles.
+  if (empty($roles) && empty($input['roles']) && !empty($form['account']['roles']['#default_value'])) {
+    $roles = $form['account']['roles']['#default_value'];
+  }
+  $roles = array_merge([AccountInterface::AUTHENTICATED_ROLE], $roles, (!empty($input['roles']) ? array_values($input['roles']) : []));
+
+  try {
+    /** @var \Drupal\password_enhancements\Entity\Storage\ConstraintEntityStorageInterface $constraint_storage */
+    $constraint_storage = Drupal::entityTypeManager()->getStorage('password_enhancements_constraint');
+    $constraints = $constraint_storage->loadByRoles($roles);
+  }
+  catch (Exception $e) {
+    Drupal::service('logger.password_enhancements')->exception($e);
+    $constraints[] = new Constraint([
+      'type' => 'error',
+      'required' => TRUE,
+      'descriptionSingular' => t('An error occurred, it is currently not possible to validate the password.'),
+    ], 'password_enhancements_constraints');
+  }
+
+  $processed_constraints = [];
+  $libraries = [
+    'password_enhancements/requirements',
+  ];
+  /** @var \Drupal\password_enhancements\PasswordConstraintPluginManager $plugin_manager */
+  $plugin_manager = Drupal::service('plugin.manager.password_constraint');
+  $plugin_config = [];
+  foreach ($constraints as $id => $constraint) {
+    // Replace dot in the constraint's ID with a dash to make it more JS/CSS
+    // friendly.
+    $id = str_replace('.', '-', $constraint->id());
+
```

Comment:
Unnecessary new line near here.

Reply:
Nope, that was on purpose. The comment belongs to the `$id` only and for the `$type` it does not, I wanted to separate it.